src/librustdoc/html/render/mod.rs

```diff
@@ -1627,9 +1610,17 @@ fn render_impl(
             );
         }
     }
-    if !impl_items.is_empty() {
+    if !impl_items.is_empty() || !default_impl_items.is_empty() {
+        let need_tags = trait_.is_some() && !impl_items.is_empty();
         w.write_str("<div class=\"impl-items\">");
+        w.push_buffer(default_impl_items);
+        if need_tags {
+            w.write_str("<details class=\"undocumented\"><summary></summary>")
+        }
         w.push_buffer(impl_items);
+        if need_tags {
+            close_tags.insert_str(0, "</details>");
```

Comment:
Can you move this up to line 1618? That way we don't need two different `if` statements for the same bool. If was wrote the open tag, we'll also set ourselves up to make sure we write the close tag later, by putting it in `close_tags`. That also allows us to eliminate the `need_tags` variable. We can write `if trait_.is_some() && !impl_items.is_empty()` directly, since we only need to write it once.

Reply:
Wow indeed! I wonder how I didn't see that before... Thanks!